meteorite_fall.js

```diff
@@ -2,37 +2,52 @@ const canvas = document.getElementById("canvas");
 const context = canvas.getContext("2d");
 const canvasHeight = canvas.height;
 const canvasWidth = canvas.width;
-const meteoriteDisplacement = 5;
+const meteoriteFallSpeed = 5;
+const meteoriteWidth = 10;
+const meteoriteHeight = 40;
 
-let meteoriteXPosition = Math.floor(Math.random() * canvasWidth); 
 let meteoriteYPosition = 0;
 
-let meteorite = {
-    width: 10,
-    height: 40
-};
-
+let meteorites = new Array();
 let player = new Player(canvasWidth / 2, canvasHeight - 10, 10, 10, 10, "blue");
 
 function main() {
-    setInterval(meteoriteFall, 50);
+    fillMeteorites();
+    setInterval(startGame, 50);
 }
 
-function drawMeteorite() {
-    context.beginPath();
-    context.rect(meteoriteXPosition, meteoriteYPosition, meteorite.width, meteorite.height)
-    context.fillStyle = "black";
-    context.fill();
-    context.closePath();
-}
+// main game functionality
+function startGame() {
+    var randomMeteoriteIndex = Math.floor(Math.random() * meteorites.length);
 
-// main game loop
-function meteoriteFall() {
     context.clearRect(0,0, canvasWidth, canvasHeight);
-    drawMeteorite();
-    player.draw(context);
 
-    meteoriteYPosition += meteoriteDisplacement;
+    if (meteorites[randomMeteoriteIndex].isFalling == false) {
```

Comment:
This entire algorithm is entirely overcomplicated in my opinion. Sorry. What you do is:
- populate an array of x amount of meteorites
- Every execution of the main game function (every 50ms) you randomly select one to start falling

What I would do is simply:
- Start with an empty array to store meteorites
- every x amount of ms, we spawn another meteorite, adding it to the array. You could even use another setInterval() for this.
- every meteorite that exists in the array will be falling. So no need for an isFalling attribute
- When a meteorite hit's the bottom of the canvas, we remove it from the array and delete it

Besides being overly complicated your algo has the following downsites:
- we have to set a hard limit to the amount of meteorites we can have in the game
- The ramp up of meteorite spawning will slow down, as more and more meteorites are already falling

Reply:
1) How does one expect to delete a meteorite? -> I had already tried setting each newly created object to null but an exception is thrown as a result.
2) We don't actually need to create new meteorites, they can be recycled by resetting the y-value. The number of "object meteorites" created will always be the same. No danger of memory leaks per say. -> I believe its more optimal to work with variables on the same object (isFalling, y-position) then to always keep creating new objects. 
3) "The ramp up of meteorite spawning will slow down, as more and more meteorites are already falling" -> I'm not entirely sure what you mean by this. However, I verified to ensure that the number of meteorites will always be the same throughout the game; there is no accumulation happening whatsoever. I ensured that y-value won't continue increasing but is instead just being reset every time a meteorite gets to the bottom of the canvas.